### PR TITLE
Fixes #938 : Adds Custom Color Badges

### LINKED ISSUE #938: Generating Custom Colored Badges
**I'm submitting a ...**
- [ ] bug report
- [X] feature request

Backend must generate custom background colored badges.

**Yes I am sending a PR for the same**

### PR DESCRIPTION
**Pull request by gabru-md**
Adds Custom Color Badges

Fixes #938 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] I have added necessary documentation (if appropriate)

### Preview Link 

- **Replace XXX with your PR no**
- Link to live demo: http://pr-XXX-fossasia-badgeyay.surge.sh  

#### Changes proposed in this pull request:

- Adds route for generating custom colored backgrounds

### CODE REVIEW

**Comment by gabru-md** [comment]
@djmgit @yashLadha @ParthS007 please review and merge asap

**Comment by gabru-md** [comment]
s2
s1

These are the screenshots showing the working of the API.
@yashLadha please check the incoming data to be accepted by the API.

@djmgit @ParthS007 @yashLadha please review and merge asap

File: api/controllers/fileUploader.py [hunk 1]
```diff
@@ -136,3 +137,34 @@ def upload_default():
     file_upload = File(filename=imageName, filetype='image', uploader=fetch_user)
     file_upload.save_to_db()
     return jsonify(DefImageSchem().dump(file_upload).data)
+
+
+@router.route('/background_color', methods=['POST'])
+def background_color():
+    try:
+        data = request.get_json()['data']
```
Comment (yashLadha) [inline api/controllers/fileUploader.py]: Can you please convert it to `request.get_json()['data']['attributes']` else all look fine to me

**Comment by gabru-md** [comment]
@yashLadha @ParthS007 @djmgit changes made. review and merge?

File: api/controllers/fileUploader.py [hunk 2]
```diff
@@ -136,3 +137,34 @@ def upload_default():
...
+    svg2png = SVG2PNG()
+
+    bg_color = '#' + str(bg_color)
+    user_defined_path = svg2png.do_svg2png(1, bg_color)
+    with open(user_defined_path, "rb") as image_file:
+        image_data = base64.b64encode(image_file.read())
+        os.remove(user_defined_path)
+
+    try:
+        imageName = saveToImage(imageFile=image_data, extension=".png")
```
Comment (yashLadha) [inline api/controllers/fileUploader.py]: Convert the `image_data` to `image_data.decode('utf-8')`

File: api/helpers/uploads.py [hunk 3]
```diff
@@ -14,8 +14,10 @@ def saveToImage(imageFile=None, extension='.png'):
 
     if not os.path.isdir(imageDirectory):
         os.makedirs(imageDirectory)
-
-    imageFile = imageFile.replace('data:image/png;base64,', '')
+    try:
+        imageFile = imageFile.replace('data:image/png;base64,', '')
+    except Exception:
```
Comment (yashLadha) [inline api/helpers/uploads.py]: No need of exception then

**Pull request merged**